src/app/hooks/useViewTracker/index.test.jsx

```diff
@@ -77,63 +111,123 @@ describe('Expected use', () => {
     url: 'http://www.bbc.com/pidgin/tori-51745682',
   };
 
-  it('should return a ref used for tracking', async () => {
-    setIntersectionNotObserved();
+  it('should return a function that can be assigned to an element to observe for intersections', async () => {
+    const { result } = renderHook(() => useViewTracker(trackingData), {
+      wrapper,
+    });
+    const element = document.createElement('div');
+
+    await result.current(element);
+
+    const { observe } = getObserverInstance(element);
+
+    expect(observe).toHaveBeenCalledWith(element);
+  });
 
+  it('should not send event to ATI when element is not in view', async () => {
     const { result } = renderHook(() => useViewTracker(trackingData), {
       wrapper,
       initialProps: {
         pageData: fixtureData,
       },
     });
+    const element = document.createElement('div');
+
+    await result.current(element);
+
+    const observerInstance = getObserverInstance(element);
+
+    act(() => {
+      triggerIntersection({
+        changes: [{ isIntersecting: false }],
+        observer: observerInstance,
+      });
+    });
 
-    expect(result.current).toBe(elementRef);
+    act(() => {
+      jest.advanceTimersByTime(1100);
+    });
+
+    expect(global.fetch).not.toHaveBeenCalled();
   });
 
-  it('should not send event to ATI when element is not in view', async () => {
-    setIntersectionNotObserved();
+  it('should skip initialising IntersectionObserver when eventTracking toggle is disabled', async () => {
```

Comment:
Just wonder if this is testing internal implementation details a bit too much, this might be give a bit more weight to extracting out the IntersectionObserver logic to its own component?

Reply:
It is a little yeh however I think it's important to test we are not initialising intersection observers when we don't need to because they do come with a performance hit.

Reply:
makes sense 👍